lightning/src/ln/channelmanager.rs

```diff
@@ -7459,48 +7497,60 @@ where
 							}
 						}
 					}
-					for (htlc_source, htlc) in monitor.get_all_current_outbound_htlcs() {
-						if let HTLCSource::PreviousHopData(prev_hop_data) = htlc_source {
-							let pending_forward_matches_htlc = |info: &PendingAddHTLCInfo| {
-								info.prev_funding_outpoint == prev_hop_data.outpoint &&
-									info.prev_htlc_id == prev_hop_data.htlc_id
-							};
-							// The ChannelMonitor is now responsible for this HTLC's
-							// failure/success and will let us know what its outcome is. If we
-							// still have an entry for this HTLC in `forward_htlcs` or
-							// `pending_intercepted_htlcs`, we were apparently not persisted after
-							// the monitor was when forwarding the payment.
-							forward_htlcs.retain(|_, forwards| {
-								forwards.retain(|forward| {
-									if let HTLCForwardInfo::AddHTLC(htlc_info) = forward {
-										if pending_forward_matches_htlc(&htlc_info) {
-											log_info!(args.logger, "Removing pending to-forward HTLC with hash {} as it was forwarded to the closed channel {}",
-												log_bytes!(htlc.payment_hash.0), log_bytes!(monitor.get_funding_txo().0.to_channel_id()));
-											false
+					for (htlc_source, (htlc, preimage_opt)) in monitor.get_all_current_outbound_htlcs() {
+						match htlc_source {
+							HTLCSource::PreviousHopData(prev_hop_data) => {
```

Comment:
I was wondering if we could apply any invariant w.r.t `option` in this case... Should it be `assert(is_none)` or something? It would be better than just dropping it on the floor in this match condition.

Reply:
Ahhh i meant `match`. Inside the `PreviousHopData` condition, i wondered if anything could be asserted about `preimage_opt` instead of just dropping it of the floor. This is totally not a blocker, but if that's possible it just helps to understand better what and where is coming from